src/runtime/runtime-debug.cc

```diff
@@ -1236,20 +1236,39 @@ struct AssertionSite {
   int source_position_;
   std::string location_;
 };
-typedef std::vector<AssertionSite> AssertionSiteVector;
+typedef std::vector<AssertionSite*> AssertionSiteVector;
 static AssertionSiteVector* gAssertionSites;
 static base::Mutex* gAssertionSitesMutex;
 
 int RegisterAssertValueSite(const std::string& desc, int source_position) {
   base::MutexGuard lock(gAssertionSitesMutex);
 
   int index = (int)gAssertionSites->size();
-  gAssertionSites->push_back({ desc, source_position, "" });
+  gAssertionSites->push_back(new AssertionSite({ desc, source_position, "" }));
   return index + BytecodeSiteOffset;
 }
 
+static inline AssertionSite& GetAssertValueSite(int32_t index) {
+  index -= BytecodeSiteOffset;
+
+  base::MutexGuard lock(gAssertionSitesMutex);
+
+  CHECK(gAssertionSites && (size_t)index < gAssertionSites->size());
+  AssertionSite* site = (*gAssertionSites)[index];
+  return *site;
+}
+
 extern std::string RecordReplayBasicValueContents(Handle<Object> value);
 
+static inline bool IgnoreAssertValueLocation(const std::string& location) {
+  static const char* pattern = getenv("RECORD_REPLAY_JS_ASSERTS_PATTERN");
+  if (!pattern) {
+    return false;
+  }
+  bool match = !!strstr(location.c_str(), pattern);
+  return !match;
+}
```

Comment:
This change adds support for only adding JS value assertions in specific sources.